packages/gatsby/src/db/loki/index.js

```diff
@@ -0,0 +1,120 @@
+const _ = require(`lodash`)
+const fs = require(`fs-extra`)
+const path = require(`path`)
+const loki = require(`@moocar/lokijs`)
+const uuidv4 = require(`uuid/v4`)
+
+// Loki is a document store with the same semantics as mongo. This
+// means there are no tables or relationships. Just a bunch of
+// collections, each with objects.
+//
+// Gatsby stores nodes in collections by splitting them up by their
+// `node.internal.type`. All nodes of a particular type go in 1
+// collection. The below `colls` object contains the metadata for
+// these collections, and the "meta collections" used to track them.
+//
+// You won't use these directly. They are used by the collection
+// functions in `./nodes.js`. E.g `getTypeCollName()` and
+// `getNodeTypeCollection`
+const colls = {
+  // Each object has keys `id` and `typeCollName`. It's a way of
+  // quickly looking up the collection that a node is contained in.
+  // E.g { id: `someNodeId`, typeCollName: `gatsby:nodeType:myType` }
+  nodeMeta: {
+    name: `gatsby:nodeMeta`,
+    options: {
+      unique: [`id`],
+      indices: [`id`],
+    },
+  },
+  // The list of all node type collections. Each object has keys
+  // `type` and `collName` so you can quickly look up the collection
+  // name for a node type.
+  // e.g { type: `myType`, collName: `gatsby:nodeType:myType` }
```

Comment:
Technically, we could get away with something like this:

```
const coll = db.getCollection(nodesDb.makeTypeCollName(typeName))
```

But this ties the name of the node type collection to the `makeTypeCollName` function. What if that function produces a different result in future, but the original collection still exists in the db? There also might be a small performance cost since we would need to calculate the string name for each lookup, vs a fast in-memory db lookup.